src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+
+    class Node<A> {
+        private A value;
+        private Node<A> next;
+        private Node<A> prev;
+
+        public Node(Node<A> prev, A value, Node<A> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+
+        public A getValue() {
+            return value;
+        }
+
+        public void setValue(A value) {
+            this.value = value;
+        }
+
+        public Node<A> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<A> next) {
+            this.next = next;
+        }
+
+        public Node<A> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<A> prev) {
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+        }
+        tail = node;
```

Comment:
refactor to avoid code duplication
```suggestion
        linkTail(value);
```